source/ada/lsp-ada_handlers-refactor_move_parameter.adb

```diff
@@ -0,0 +1,318 @@
+------------------------------------------------------------------------------
+--                         Language Server Protocol                         --
+--                                                                          --
+--                        Copyright (C) 2021, AdaCore                       --
+--                                                                          --
+-- This is free software;  you can redistribute it  and/or modify it  under --
+-- terms of the  GNU General Public License as published  by the Free Soft- --
+-- ware  Foundation;  either version 3,  or (at your option) any later ver- --
+-- sion.  This software is distributed in the hope  that it will be useful, --
+-- but WITHOUT ANY WARRANTY;  without even the implied warranty of MERCHAN- --
+-- TABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public --
+-- License for  more details.  You should have  received  a copy of the GNU --
+-- General  Public  License  distributed  with  this  software;   see  file --
+-- COPYING3.  If not, go to http://www.gnu.org/licenses for a complete copy --
+-- of the license.                                                          --
+------------------------------------------------------------------------------
+
+with Ada.Exceptions;
+
+with Ada.Strings.UTF_Encoding;
+
+with Langkit_Support.Text;
+
+with Libadalang.Analysis; use Libadalang.Analysis;
+
+with Laltools.Common; use Laltools.Common;
+
+with LSP.Messages;
+with LSP.Messages.Client_Requests;
+with LSP.Lal_Utils;
+
+with VSS.Strings;
+with VSS.Strings.Conversions;
+
+package body LSP.Ada_Handlers.Refactor_Move_Parameter is
+
+   ------------------------
+   -- Append_Code_Action --
+   ------------------------
+
+   procedure Append_Code_Action
+     (Self              : in out Command;
+      Context           : Context_Access;
+      Commands_Vector   : in out LSP.Messages.CodeAction_Vector;
+      Target_Subp       : Libadalang.Analysis.Basic_Decl;
+      Parameter_Index   : Positive;
+      Move_Direction    : Move_Direction_Type)
+   is
+      Pointer : LSP.Commands.Command_Pointer;
+      Code_Action : LSP.Messages.CodeAction;
+      Where       : constant LSP.Messages.Location :=
+        LSP.Lal_Utils.Get_Node_Location (Target_Subp.P_Defining_Name.F_Name);
+
+      function Image
+        (D : Move_Direction_Type)
+         return Langkit_Support.Text.Text_Type
+      is ((if D = Right then "right" else "left"));
```

Comment:
Is it `right` and `left` good for moving "direction" description?
Personally, I almost never have two parameters in line, but one under another.
Could more general (`before`/`after` or `forward`/`backward`) be better for this?

Reply:
Agree.